Feature/settings - Add url settings for getting the swagger ready to use

elliotyim opened the pull request:
drf-yasg를 사용하는 데 있어서 누락된 세팅을 추가합니다.
이 세팅을 추가하고 난 뒤에 urls.py에 연결되어있는 페이지(ex: localhost:8000/swagger/)에 들어가면 사용가능한 api 정보를 전부 볼 수 있습니다.
(drf를 사용하여 api를 만들었다면 자동으로 api를 문서로 만들어줍니다.)

기본적인 설명은 모듈 내에서 docstring으로 작성해두면 api 설명란에 자동으로 추가해줍니다.
decorator를 사용하여 api에 추가로 상세한 정보를 제공하려면 아래의 링크를 참조하세요.
Link: https://drf-yasg.readthedocs.io/en/stable/custom_spec.html#support-for-serializermethodfield

choitaehoon1998 committed:
Update project settings

Change requirements.txt to use libary for this project
Change settings.py for this project

choitaehoon1998 committed:
Change settings

choitaehoon1998 committed:
Change "settings.py" to use environ varialbe

choitaehoon1998 committed:
Rebase develop branch

choitaehoon1998 committed:
Remove Unrelated file

choitaehoon1998 committed:
After Merge

elliotyim committed:
Add url settings for getting the swagger ready to use

elliotyim committed:
Add drf-yasg stubs for mypy

elliotyim committed:
Change model-mommy into model-bakery

Model mommy is deprecated, do not use it.

elliotyim committed:
Add mypy ignoring settings

Removed:
- model mommy

Added:
- model bakery
- drf yasg stubs

elliotyim committed:
Fix setup.cfg to prevent mypy error

Removed:
- drf-yasg-stubs

Added:
- drf_yasg
- ruamel

hyeonjames commented on dev_up/urls.py:
```diff
@@ -15,7 +15,23 @@
 from django.urls import path
+from drf_yasg import openapi
+from drf_yasg.views import get_schema_view
+from rest_framework import permissions  # type: ignore
```
djangorestframework-stubs 넣고, `# type: ignore` 지워주세요

elliotyim replied:
우선 djangorestframework-stubs는 다른 곳들에도 사용될 것 같아서
requirements.txt에 추가했습니다.
또한 permissions는 밑의 permission_classes의 값으로서 사용되고 있었지만
이는 default 값으로 지정되어 있기에 중복되기도 하고 mypy 오류를 일으키기에 제거하였습니다.
따라서 rest_framework의 permissions 또한 필요없을 것이라 판단되어 제거하였습니다.

hyeonjames commented on requirements.txt:
```diff
@@ -8,6 +8,7 @@ mypy
 django-stubs
 pytest-django
 djangorestframework
```
이거 버전 명시하는게 좋을 것 같아요

elliotyim replied:
넵 수정하였습니다.

hyeonjames commented on requirements.txt:
```diff
@@ -8,6 +8,7 @@ mypy
-model_mommy
 drf-yasg==1.17.1
+drf-yasg-stubs==0.1.0
+psycopg2
```
마찬가지로 버전 명시 하는게 좋겠어요ㅕ

elliotyim replied:
넵 수정하였습니다.

elliotyim committed:
Specify versions of dependency: requirements.txt

Change Log:
- Added djangorestframework-stubs
- Specified versions of a list of dependency below
  - django-stubs
  - pytest-django
  - djangorestframework
  - psycopg2

elliotyim committed:
Remove redundancy

As the default value of permission_classes is permissions.AllowAny,
removed the line that specify permission classes on default swagger urls.

If `permissions.AllowAny` is set, it will cause a mypy error
: error: Argument "permission_classes" to "get_schema_view" has incompatible type
  "Tuple[Type[AllowAny]]"; expected "Optional[Tuple[str]]"

This, `Optional[Tuple[str]], means: str 'rest_framework.permissions.AllowAny'
in settings.py of `rest_framework` that is redundancy.

elliotyim committed:
Remove drf-yasg-stubs from requirements.txt

This stubs are unstable at the moment.

elliotyim committed:
Update version of django-stubs

mypy 0.770 that we are currently using does not compatible with django-stubs 1.4.0

hyeonjames reviewed (approved):
수고하셨어요 ~ 👍

choitaehoon1998 commented:
수고하셨습니당 👍

choitaehoon1998 committed:
Change setup.cfg

choitaehoon1998 committed:
Merge branch 'feature/settings' of https://github.com/devv-up/backend into feature/settings
